src/item.cpp

```diff
@@ -1584,7 +1585,136 @@ bool item::stacks_with( const item &rhs, bool check_components, bool combine_liq
             return false;
         }
     }
-    return contents.stacks_with( rhs.contents, depth, maxdepth );
+    return true;
+}
+
+bool _stacks_variant( item const &lhs, item const &rhs )
+{
+    return ( !lhs.has_itype_variant() && !rhs.has_itype_variant() ) ||
+           ( lhs.has_itype_variant() && rhs.has_itype_variant() &&
+             lhs.itype_variant().id == rhs.itype_variant().id );
+}
+
+bool _stacks_components( item const &lhs, item const &rhs, bool check_components )
+{
+    return ( !check_components && !lhs.is_comestible() && !lhs.is_craft() ) ||
+           // Only check if at least one item isn't using the default recipe or is comestible
+           ( lhs.components.empty() && rhs.components.empty() ) ||
+           ( lhs.get_uncraft_components() == rhs.get_uncraft_components() );
+}
+
+} // namespace
+
+stacking_info item::stacks_with( const item &rhs, bool check_components, bool combine_liquid,
+                                 bool check_cat, int depth, int maxdepth ) const
+{
+    tname::segment_bitset bits;
+    if( type == rhs.type ) {
+        bits.set( tname::segments::TYPE );
+        bits.set( tname::segments::ENGINE_DISPLACEMENT );
+        bits.set( tname::segments::WHEEL_DIAMETER );
+        bits.set( tname::segments::WHITEBLACKLIST, _stacks_whiteblacklist( *this, rhs ) );
+    }
+
+    if( !check_cat && bits.none() ) {
+        return {};
+    }
+
+    bits.set( tname::segments::CATEGORY,
+              !check_cat ||
+              get_category_of_contents( depth, maxdepth ).get_id() ==
+              rhs.get_category_of_contents( depth, maxdepth ).get_id() );
+
+    if( check_cat && bits.none() ) {
+        return {};
+    }
+
+    bits.set( tname::segments::VARIANT, _stacks_variant( *this, rhs ) );
+    bits.set( tname::segments::RELIC, is_same_relic( rhs ) );
+
+    const std::set<ammotype> ammo = ammo_types();
+    bits.set( tname::segments::MONEY,
+              !is_money( ammo ) || ( ammo_remaining( ammo ) == rhs.ammo_remaining( ammo ) ) );
```

Comment:
It turns out that this check isn't needed at all since money sits in a magazine now and the contents check covers it. The test `stacking_cash_cards` confirms it.

I'm removing it since it's really expensive and not worth aggregating.